token-handler/src/common/storage.rs

```diff
@@ -4,4 +4,7 @@ use multiversx_sc::imports::SingleValueMapper;
 pub trait CommonStorage {
     #[storage_mapper]
     fn sov_prefix(&self) -> SingleValueMapper<ManagedBuffer>;
+
+    #[storage_mapper]
+    fn header_verifier_address(&self) -> SingleValueMapper<ManagedAddress>;
```

Comment:
In header-verifier we have a condition that only enshrine is able to call remove_executed_hash, so this storage needs to be in enshrined.

We can change the implementation that only token-handler can remove_executed_hash, but we need to analyse if this is safe. Anyways this would be another PR

Reply:
@sasurobert do you think that `token-handler` should remove the executed hashes from the `header-verifier` or keep that role to `enshrine-esdt` ?

Reply:
I would say to not complicate `token-handler` contract with whitelist header verifier contracts and stuff. 
`enshrine-esdt` is verifying hash is present in `header_verifier` so enshrine will also remove it in the end.

Reply:
i agree with octavian.